### DISCUSSION TITLE
Is multithreading (multiple core) possible with Tokio ?

### DISCUSSION THREAD

**Discussion by T-berton** [tokio-rs/tokio#4551] [Q&A] [answered]
Hi Everyone,
Here my situation, i am working on a Raspberry Pi 3 where i have 3 sensors on it. I did multi-tasking on these sensors with Tokio and it worked very nice ! Indeed, I have my functions on my 3 sensors that run asynchronously. I would like to do now multithreading on these sensors. I mean i would like that each core from my CPU is linked to one sensor. 
Is it possible to do it with Tokio or do i have to do it with a RTOS ?
Thanks a lot for your help.

**Comment by asonix** [comment, marked as answer]
By default tokio makes use of as many threads as you have cores on your computer for asynchronous tasks, so on a raspberry pi 3, that would be 4 threads. Is there anything specific you want to get out of multi-threading, or is this just a personal exercise

**Reply by T-berton** [reply]
Oh okay i understand, thanks. Hmmm it's more a personnal exercice, i would like to have better performance by doing multi-threading.

**Comment by Noah-Kennedy** [comment]
What sensors are we talking about, and what are you hoping to get out of multithreading?

**Reply by T-berton** [reply]
Actually i am using UWB modules to do triangulation algorithm positionning. So I have three UWB on my raspberry PI and i would like each uwb is used by each core in order to have better performance than using only one core.

So if i understand well, tokio will use automatically my 4 core from my raspberry pi ?
Thanks a lot for your answer.

**Reply by Noah-Kennedy** [reply]
I would recommend *not* doing any blocking CPU work in tokio. Tokio is optimized for asynchronous, nonblocking code. You should instead spawn separate threads for this processing work.

**Reply by Noah-Kennedy** [reply]
What exactly do you need async for here?

**Reply by Noah-Kennedy** [reply]
Do you have networking that needs to be async?
